### PR TITLE
fix #104 list unique cities user have in his trip

### LINKED ISSUE #104: [API Request] API to get a list of all the unique cities user have in  his trips
A GET list of all the cities any user has in his list of trips.
If any user has 3 trips in 2 cities, output should be a list of 2 cities in the alphabetical order of city names
### Input 
+ user id
### Output
List of cities in alphabetical order of city names

Reference PR: #69

### PR DESCRIPTION
**Pull request by mohitkh7**
# Description

list down all the cities any user has in his list of trips.

API url : `/api/get-trip-cities/<user_id>`

Sample Output :
`{
    "cities": [
        "Indore",
        "Kanpur"
    ]
}`

Fixes #104 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `flake8`
- [x] `python manage.py test`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: api/modules/trips/views.py [hunk 1]
```diff
@@ -237,3 +237,26 @@ def remove_user_from_trip(request, trip_id):
...
+    # if user with user_id does not exist
+    if not User.objects.filter(id=user_id).exists():
+        error_message = "User does not exists."
+        return Response(error_message, status=status.HTTP_404_NOT_FOUND)
+
+    city_list = []
+    trips = Trip.objects.filter(users=user_id)
+    for trip in trips:
+        city_list.append(trip.city.city_name)
+    city_list = sorted(list(set(city_list)))
```
Comment (prabhakar267) [inline api/modules/trips/views.py]: You can use `AllCitiesSerializer` instead of sending just the list of names

**Review by prabhakar267 — CHANGES_REQUESTED** [review]

**Comment by mohitkh7** [comment]
@prabhakar267 requested changes done :+1:

File: api/modules/trips/views.py [hunk 2]
```diff
@@ -237,3 +238,27 @@ def remove_user_from_trip(request, trip_id):
...
+    """
+    # if user with user_id does not exist
+    if not User.objects.filter(id=user_id).exists():
+        error_message = "User does not exists."
+        return Response(error_message, status=status.HTTP_404_NOT_FOUND)
+
+    city_id_list = []
+    trips = Trip.objects.filter(users=user_id)
+    for trip in trips:
+        city_id_list.append(trip.city.id)
```
Comment (prabhakar267) [inline api/modules/trips/views.py]: You can append `trip.city` instead of making a list of city IDs and then fetching each city by ID

**Review by prabhakar267 — CHANGES_REQUESTED** [review]

File: api/modules/trips/views.py (hunk 2 above)
Comment (mohitkh7) [reply api/modules/trips/views.py]: :+1:

**Review by prabhakar267 — APPROVED** [review]

**Pull request merged**